### DISCUSSION TITLE
USB Drive Stuck on Eject w/Docker

### DISCUSSION THREAD

**Discussion by georgecgibson** [automatic-ripping-machine/automatic-ripping-machine#1201] [Q&A]
Hello - recently setup ARM via Docker with two USB drives. One is a 5.25 unit in an enclosure and works as you'd expect. The other however works great up until it's time to eject, either manually or via the ARM process. It will successfully eject, but then continue to eject without stopping. I made the mistake of hoping it may have a timeout last night (my fault for being that optimistic/lazy) and it was still trying this morning. _Ouch._

The only thing I've found to stop this behavior is to restart the ARM docker container. Obviously that's not ideal. 

I need some information as to where I should be looking to see if I can figure out why this is happening. Glancing at the existing logs that I have available I didn't see anything related to the eject process. I noted the info in the wiki about empty.log, but I don't see that in the logs directory despite the presence of the other logs. 

Model of the drive is Pioneer BDR-XS07UHD. This is also my only UHD capable drive at the moment. 

Any pointers/help is appreciated. Thanks!

**Comment by Oddadin** [comment]
Same issue but with an external Verbatim drive. Whilst I've not solved this issue myself I have found a workaround for it with using the following command:

sudo setcd -f0 /dev/sr0

When you run this (need to have setcd installed too)  it should set the "Use O_NONBLOCK flag" to cleared. The drive then works as it should until the next restart. 

Can't see anything in the logs that points to a culprit and I've got my setup set to just backup the disc and to not auto-eject.
